plugins/clients/apps-client/src/hopeit/apps_client/__init__.py

```diff
@@ -394,11 +397,21 @@ def _auth_headers(self, context: EventContext, now_ts: int):
 
     async def _parse_response(self, response, context: EventContext,
                               datatype: Type[EventPayloadType],
-                              target_event_name: str) -> List[EventPayloadType]:
+                              target_event_name: str,
+                              responses: Optional[Dict[int, Type[EventPayloadType]]]) -> List[EventPayloadType]:
         """
         Parses http response from external App, catching Unathorized errors
         and converting the result to the desired datatype
         """
+        if responses and response.status in responses:
+            if response.content_type == 'text/plain':
+                data = {target_event_name: await response.text()}
+            else:
+                data = await response.json()
+            response_type = responses[response.status]
+            if isinstance(data, list):
+                return Payload.from_obj(data, list, item_datatype=response_type)  # type: ignore
+            return [Payload.from_obj(data, response_type, key=target_event_name)]
         if response.status == 200:
```

Comment:
@leosmerling-hopeit what do you think about first trying on `responses` instead of `response.status == 200`? 
PROS:
This allows listing all expected response types in one place, and if you are in favor of this idea switch to optional the `datatype` parameter?
CONS: 
if switch in favor of optional `datatype`, `responses` already it is, and non of two are provided, the call will fail.

Reply:
Use reponses as a dict, adding 200 status to it, then use just the `response` dict to parse the response using the right datatype. Then capture KeyError exception to handle 401, 500 and other unhandled statuses. See example above.